UnityProject/Assets/Scripts/Shuttles/MatrixMove.cs

```diff
@@ -9,314 +9,192 @@
-	public MatrixState ClientState => clientState;
-
-	///client's transform, can get dirty/predictive
-	private MatrixState clientState = MatrixState.Invalid;
 
+	[SyncVar(hook = nameof(SyncInitialPosition))]
+	private Vector3 initialPosition;
 	/// <summary>
-	/// Tracks whether we've received our initial rotation offset for this matrix.
+	/// Initial position for offset calculation, set on start and never changed afterwards
 	/// </summary>
-	public bool ReceivedInitialRotation { get; private set; } = false;
+	public Vector3Int InitialPosition => initialPosition.RoundToInt();
+
+	[SyncVar(hook = nameof(SyncPivot))]
+	private Vector3 pivot;
 	/// <summary>
-	/// Tracks whether we are performing our initial rotation offset based on the initial rotation
-	/// we received.
+	/// local pivot point, set on start and never changed afterwards
 	/// </summary>
-	private bool performingInitialRotation = false;
-
-	/// Is only present to match server's flight routines
-	private MatrixState clientTargetState = MatrixState.Invalid;
-
-	private bool isMovingClient => clientState.IsMoving && clientState.Speed > 0f;
+	public Vector3Int Pivot => pivot.RoundToInt();
 
 	/// <summary>
-	/// Does current transform rotation not yet match the target client offset, and thus this matrix's transform needs to
-	/// be rotated to match the target?
+	/// All the various events that can be subscribed to on this matrix
 	/// </summary>
-	public bool ClientNeedsRotation =>
-		Quaternion.Angle(transform.rotation, clientState.RotationOffset.Quaternion) != 0;
-
-	private bool ClientPositionsMatch => clientTargetState.Position == clientState.Position;
-
-	//editor (global) values
-	public UnityEvent OnClientStart = new UnityEvent();
-	public UnityEvent OnClientStop = new UnityEvent();
-	public UnityEvent OnClientFullStop = new UnityEvent();
-	public UnityEvent OnStart = new UnityEvent();
-	public UnityEvent OnStop = new UnityEvent();
+	public readonly MatrixMoveEvents MatrixMoveEvents = new MatrixMoveEvents();
 
+	//server-only values
+	public MatrixState ServerState => serverState;
+	public bool IsMovingServer => serverState.IsMoving && serverState.Speed > 0f;
+	//client-only values
+	public MatrixState ClientState => clientState;
+	private MatrixInfo matrixInfo;
+	public MatrixInfo MatrixInfo => matrixInfo;
+	private ShuttleFuelSystem shuttleFuelSystem;
+	public ShuttleFuelSystem ShuttleFuelSystem => shuttleFuelSystem;
 	/// <summary>
-	/// Offset of the most recent rotation that occurred, used so we can
-	/// save this info for firing the OnRotateEnd event,
+	/// Gets the rotation offset this matrix has from its initial mapped
+	/// facing.
 	/// </summary>
-	private RotationOffset previousRotation;
-
+	public RotationOffset FacingOffsetFromInitial => ClientState.FacingOffsetFromInitial(this);
 
 	/// <summary>
 	/// If it is currently fuelled
 	/// </summary>
+	[NonSerialized]
 	public bool IsFueled;
 
 	/// <summary>
 	/// Requires Fuel
 	/// </summary>
+	[NonSerialized]
```

Comment:
why is it NonSerialized now?
some ships do require fuel and some don't, but `true` should be default

Reply:
I set it to false in case people were trying to map stuff in and they can't work out why it doesn't want to move